crane_x7_tests/launch/pick_and_place.launch.py

```diff
@@ -0,0 +1,148 @@
+                    ]
+                )
+            ]
+        ),
+    )
+
+    demo = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            [
+                PathJoinSubstitution(
+                    [
+                        FindPackageShare("crane_x7_examples"),
+                        "launch",
+                        "example.launch.py",
+                    ]
+                )
+            ]
+        ),
+        launch_arguments={
+            "use_sim_time": "true", 
+            "example": "pick_and_place"
+        }.items(),
+    )
+
+    
+    return LaunchDescription(
+        [
+            sim,
+            TimerAction(period=15.0, actions=[demo]),  # Need to optimize period, if under 10 seconds will not run
+            launch_testing.actions.ReadyToTest(),
+        ]
+    ), {"demo": demo, "sim":sim}
+
+
+def get_model_location(model_name):
+    """
+    Function to get the location of a model
+    """
+
+    command = ["ign", "model", "-m", model_name, "-p"]
+    result = subprocess.run(command, check=True, text=True, stdout=subprocess.PIPE)
+    result_output = result.stdout
+
+    print(result_output)
+
+    # Extract position data if it exists
+    position_pattern = r"\[(.*?)\]"
+
+    # Search for all matching instances of the list pattern in the text
+    matches = re.findall(position_pattern, result_output)
+
+    if len(matches) >= 2:
+        position = matches[-2]
+
+        # Extract the individual numbers from the matched string
+        numbers = re.findall(r"[\d.-]+", position)
+        
+        if len(numbers) >= 3:
+            x_variable = float(numbers[0])
+            y_variable = float(numbers[1])
+            z_variable = float(numbers[2])
+            print(f"x box coordinate: {x_variable}")
+            print(f"y box coordinate: {y_variable}")
+            print(f"z box coordinate: {z_variable}")
+        else:
+            x_variable = 0.0
+            y_variable = 0.0
+            z_variable = 0.0
+    else:
+        print("No x, y, or z variables found in the text.")
+
+    return x_variable, y_variable, z_variable
+
+class TestPickAndPlace(unittest.TestCase):
+    def test_box_moved(self, proc_output):
+        """
+        Test case to see if box has moved, if the values are no longer equal it means that the block has shifted position
+        """
+
+        # proc_output.assertWaitFor("Node Is Ready", timeout=180)
+        decimal = 3
+        model_name = "wood_cube_5cm"
+
+        (
+            x_variable,
+            y_variable,
+            z_variable,
+        ) = get_model_location(model_name)
+
+        # proc_output.assertWaitFor("Plan and Execute request complete!", timeout=180)
+        sleep(40)
+
+        (
+            x_new_variable,
+            y_new_variable,
+            z_new_variable,
+        ) = get_model_location(model_name)
+
+        self.assertNotEqual(x_variable, x_new_variable, decimal)
```

Comment:
Since we are not doing a "push" test, it may be better to test that the cube ends up at/near the desired place location.  Because if for some reason, the cube gets picked up but falls from the gripper on the way, we want that to be marked as false.

Ideally, we would like  import the place location from the example (this way if the coordinates are changed in the example, the test does not break). But this may not be possible/easy. In that case, you can log the final position once, and use that for the test (maybe rounded) and compare with some reasoble margins ( 1cm?)

Reply:
Pick and place test is done. WIP: getting the final values without hard coding them in the test

Reply:
Hard coding the values is perfectly acceptable if the alternative is too complicated